### PR TITLE
manage_user

### PR DESCRIPTION
**Pull request by weilingooi**
finish manage user

### CODE REVIEW

**Comment by weilingooi** [comment]
ready

File: Gemfile [hunk 1]
```diff
@@ -42,4 +42,6 @@ end
 
 gem "config"
 gem "jquery-rails", "4.3.1"
-
+gem "faker", "1.7.3"
+gem "kaminari"
+gem "bootstrap-kaminari-views"
```
Comment (manhnt-0960) [inline Gemfile]: please add newline at end of file

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -19,4 +21,11 @@ def destroy
...
   end
-end
+
+   private
+
+  def log_in
+    return unless logged_in?
+    redirect_to user_path id: current_user.id
+  end
+end
```
Comment (manhnt-0960) [inline app/controllers/sessions_controller.rb]: similar, please check all file

**Comment by weilingooi** [comment]
ready

File: Gemfile (hunk 1 above)
Comment (saobv-2223) [reply Gemfile]: fix this, please

File: app/views/layouts/_header.html.erb [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
-	        </ul>
-	      </li>
-	    <% else %>
-	      <li><%= link_to t("log_in"), login_path %></li>
-	    <% end %>
-      </ul> 
-    </nav>
-  </div>
-</header>
+ <header class="navbar navbar-fixed-top navbar-inverse">
```
Comment (saobv-2223) [inline app/views/layouts/_header.html.erb]: indenting 2 spaces, please

File: app/views/layouts/_header.html.erb [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
+        <ul class="nav navbar-nav navbar-right"> 
+	  <li><%= link_to t(".home"), home_path %></li> 
+	  <li><%= link_to t(".help"), help_path %></li> 
+	  <% if logged_in? %>
+	    <li><%= link_to t(".user"), users_path %></li>
+	    <li class="dropdown">
+              <%= link_to ":user.id", {class: "dropdown-toggle", data: {"toggle" =>  "dropdown"}} do %>
+                <%= t(".account") %>
+                <b class="caret"></b>
+              <% end %>
```
Comment (saobv-2223) [inline app/views/layouts/_header.html.erb]: indenting only 2 spaces

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title><%= full_title yield(:title) %></title>
+    <title><%= yield(:title) %> | <%= t ".title" %></title>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb]: remove ()

File: app/views/shared/_error_messages.html.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<% if object.errors.any? %> 
+	<div id="error_explanation">
```
Comment (saobv-2223) [inline app/views/shared/_error_messages.html.erb]:
```suggestion
  <div id="error_explanation">
```

File: app/views/static_pages/help.html.erb [hunk 7]
```diff
@@ -4,6 +4,6 @@
 <p>
 Get help on the Ruby on Rails Tutorial at the
 <%= link_to "Rails Tutorial help page", Settings.links.rails_tutorial.help_page %>.
-  To get help on this sample app, see the
-<%= link_to "<em>Ruby on Rails Tutorial</em>book", Settings.links.rails_tutorial.book_page %>.
+ To get help on this sample app, see the
+<%= link_to "Ruby on Rails Tutorial Book", Settings.links.rails_tutorial.book_page %>.
```
Comment (saobv-2223) [inline app/views/static_pages/help.html.erb]: indenting 2 spaces for these lines

File: app/views/users/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+<table class="table table-bordered table-hover">
+  <thead>
+  <tr>
+    <th><%= t ".avatar" %></th>
+    <th><%= t "users.new.name" %></th>
+    <th>Email</th>
+    <% if current_user.admin? %>
+      <th><%= t ".delete" %></th>
+    <% end %>
+  </tr>
```
Comment (saobv-2223) [inline app/views/users/index.html.erb]: indenting 2 spaces

File: app/views/users/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+      <td><%= user.name %></td>
+      <td><%= user.email %></td>
+      <% if current_user.admin? %>
+        <td><%= link_to t(".delete"), user, method: :delete, data: {confirm: t(".confirm")}, class: "btn btn-danger" %></td>
+      <% end %>
+    </tr>
+  <% end %>
+  </tbody>
+</table>
+<%= paginate @user, theme: 'twitter-bootstrap-3' %>
```
Comment (saobv-2223) [inline app/views/users/index.html.erb]: using " please

File: config/application.rb [hunk 10]
```diff
@@ -11,6 +11,11 @@ class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
     config.load_defaults 6.0
 
+    config.i18n.load_path += Dir[Rails.root.join('my', 'locales', '*.{rb,yml}')] # it will load all file at locales folder and, only read rm anf yml file for i18n
```
Comment (saobv-2223) [inline config/application.rb]: using " please

File: config/locales/en.yml [hunk 11]
```diff
@@ -61,3 +76,16 @@ en:
...
+    log_in: "Log In"
+
+  views:
+    pagination:
+      first: "« FIRST"
+      last: "LAST »"
+      previous: "‹ PREV"
+      next: "NEXT ›"
+      truncate: "…"
+
```
Comment (saobv-2223) [inline config/locales/en.yml]: add endline, please

**Comment by weilingooi** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 12]
```diff
@@ -125,3 +125,21 @@ input.btn.btn-primary {
...
+  }
+
+  input.btn.btn-warning {
+    color: #fff;
+    background-color: #f0ad4e;
+    border-color: #eea236;
+    margin-top: 10px;
+}
+}
+
```
Comment (manhnt-0960) [inline app/assets/stylesheets/custom.scss]: please check indent and end of line

File: app/controllers/sessions_controller.rb [hunk 13]
```diff
@@ -19,4 +21,11 @@ def destroy
     flash[:success] = t ".logout_success"
     redirect_to login_path
   end
+
+   private
```
Comment (manhnt-0960) [inline app/controllers/sessions_controller.rb]:
```suggestion
  private
```

Comment (manhnt-0960) [reply app/controllers/sessions_controller.rb]: remove one space before "private" keyword

File: app/controllers/users_controller.rb [hunk 14]
```diff
@@ -19,18 +21,64 @@ def create
...
+      flash[:success] = t ".update_success"
+      redirect_to @user
+    else
+      flash[:warning] = t ".update_fail"
+      render :edit
+    end
+  end
+
+  def destroy
+    @user.destroy
```
Comment (manhnt-0960) [inline app/controllers/users_controller.rb]: plase handle case delete fail

File: app/helpers/sessions_helper.rb [hunk 15]
```diff
@@ -5,18 +5,27 @@ def log_in user
...
+  def current_user? user
+    user && user == current_user
+  end
+
   def logged_in?
     current_user.present?
   end
+
+  def user_form_path
+    current_user.present? ? nil : signup_path
```
Comment (manhnt-0960) [inline app/helpers/sessions_helper.rb]: signup_path if current_user.blank?

File: app/views/users/index.html.erb [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+      <th><%= t ".avatar" %></th>
+      <th><%= t "users.new.name" %></th>
+      <th>Email</th>
+      <% if current_user.admin? %>
+        <th><%= t ".delete" %></th>
+      <% end %>
+    </tr>
+  </thead>
+  <tbody>
+  <% @user.each do |user| %>
```
Comment (manhnt-0960) [inline app/views/users/index.html.erb]: please use render collection

**Comment by weilingooi** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 17]
```diff
@@ -125,3 +125,20 @@ input.btn.btn-primary {
...
+    padding: 10px 0;
+    border-bottom: 1px solid red;
+  }
+
+  input.btn.btn-warning {
+    color: #fff;
+    background-color: #f0ad4e;
+    border-color: #eea236;
+    margin-top: 10px;
+  }
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: image

File: app/views/layouts/_header.html.erb [hunk 18]
```diff
@@ -3,24 +3,24 @@
...
               <b class="caret"></b>
             <% end %>
             <ul class="dropdown-menu">
-	          <li><%= link_to t("profile"), current_user %></li>
-		      <li><%= link_to t("settings"), "#" %></li>
+	          <li><%= link_to t(".profile"), user_path(id: current_user.id) %></li>
+		      <li><%= link_to t(".settings"), edit_user_path(id: current_user.id) %></li>
 		      <li class="divider"></li>
-		      <li><%= link_to t("logout"), logout_path, method: :delete %></li>
+		      <li><%= link_to t(".logout"), logout_path, method: :delete %></li>
```
Comment (saobv-2223) [inline app/views/layouts/_header.html.erb]: indenting only 2 spaces, check again please

**Comment by weilingooi** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 19]
```diff
@@ -125,3 +127,20 @@ input.btn.btn-primary {
...
+  margin: 0;
+  li {
+    overflow: auto;
+    padding: 10px 0;
+    border-bottom: 1px solid red;
+  }
+
+  input.btn.btn-warning {
+    color: $red;
+    background-color: #background-warning;
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: check color again

File: app/controllers/users_controller.rb [hunk 20]
```diff
@@ -19,18 +21,67 @@ def create
...
-  
+
+  def index
+    @user = User.select(:id, :name, :email)
+                .page(params[:page]).per Settings.per_page
+  end
+
+  def edit; end
+
+   def update
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
  def update
```

File: app/controllers/users_controller.rb [hunk 21]
```diff
@@ -19,18 +21,67 @@ def create
...
+    flash[:danger] = t ".user_not_found "
+    redirect_to home_path
    end
-  
-  def user_params
-    params.require(:user).permit User::USERS_PARAMS
+
+  def check_login
+    return if logged_in?
+    store_location
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
    return if logged_in?
    
    store_location
```

File: app/controllers/users_controller.rb [hunk 22]
```diff
@@ -19,18 +21,67 @@ def create
...
+    flash[:warning] = t ".not_login"
+    redirect_to login_path
+  end
+
+  def correct_user
+    return if current_user? @user
+    flash[:danger] = t ".incorrect_user"
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
    return if current_user? @user
    
    flash[:danger] = t ".incorrect_user"
```

File: app/controllers/users_controller.rb [hunk 23]
```diff
@@ -19,18 +21,67 @@ def create
...
+    redirect_to root_path
+  end
+
+  def is_admin
+    return if is_admin?
+    flash[:warning] = t ".incorrect_user"
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
    return if is_admin?
    
    flash[:warning] = t ".incorrect_user"
```

File: app/views/layouts/_header.html.erb [hunk 24]
```diff
@@ -3,26 +3,26 @@
...
 	      </li>
 	    <% else %>
-	      <li><%= link_to t("log_in"), login_path %></li>
+	      <li><%= link_to t(".log_in"), login_path, method: :delete %></li>
 	    <% end %>
       </ul> 
     </nav>
   </div>
-</header>
+</header>
```
Comment (saobv-2223) [inline app/views/layouts/_header.html.erb]: add endline

**Comment by weilingooi** [comment]
ready

File: app/helpers/sessions_helper.rb [hunk 25]
```diff
@@ -5,18 +5,27 @@ def log_in user
...
+  def current_user? user
+    user && user == current_user
+  end
+
   def logged_in?
     current_user.present?
   end
+
+  def user_form_path
+    current_user.blank? ? nil : signup_path
```
Comment (manhnt-0960) [inline app/helpers/sessions_helper.rb]:
```suggestion
    signup_path if current_user.present?
```

File: db/migrate/20200820092055_add_admin_to_users.rb [hunk 26]
```diff
@@ -0,0 +1,6 @@
+class AddAdminToUsers < ActiveRecord::Migration[6.0]
+  def change
+    add_column :users, :admin, :boolean, default: false
+  end
+  end
```
Comment (manhnt-0960) [inline db/migrate/20200820092055_add_admin_to_users.rb]: remove line 5

**Comment by weilingooi** [comment]
ready

**Comment by vanvtt-0952** [comment]
merged

**Pull request merged**